rust/sdk-processor/src/parquet_processors/mod.rs

```diff
@@ -68,6 +68,55 @@ impl ParquetTypeStructs {
             ParquetTypeEnum::MoveModule => ParquetTypeStructs::MoveModule(Vec::new()),
         }
     }
+
+    pub fn get_table_name(&self) -> &'static str {
+        match self {
+            ParquetTypeStructs::MoveResource(_) => "move_resources",
+            ParquetTypeStructs::WriteSetChange(_) => "write_set_changes",
+            ParquetTypeStructs::Transaction(_) => "parquet_transactions",
+            ParquetTypeStructs::TableItem(_) => "table_items",
+            ParquetTypeStructs::MoveModule(_) => "move_modules",
+        }
+    }
+}
+
+pub trait ParquetStruct {}
+
+impl ParquetStruct for MoveResource {}
+impl ParquetStruct for WriteSetChangeModel {}
+impl ParquetStruct for ParquetTransaction {}
+impl ParquetStruct for TableItem {}
+impl ParquetStruct for MoveModule {}
+
+impl ParquetTypeStructs {
+    pub fn get_type(&self) -> ParquetTypeEnum {
+        match self {
+            ParquetTypeStructs::MoveResource(_) => ParquetTypeEnum::MoveResource,
+            ParquetTypeStructs::WriteSetChange(_) => ParquetTypeEnum::WriteSetChange,
+            ParquetTypeStructs::Transaction(_) => ParquetTypeEnum::Transaction,
+            ParquetTypeStructs::TableItem(_) => ParquetTypeEnum::TableItem,
+            ParquetTypeStructs::MoveModule(_) => ParquetTypeEnum::MoveModule,
+        }
+    }
+
+    /// Get a vector of trait object references to the inner structs
+    pub fn get_structs(&self) -> Vec<&dyn ParquetStruct> {
```

Comment:
interesting, were you able to get the compiler to accept dynamic trait objects?

Reply:
actually no lol, will remove this!

Reply:
wouldn't you need to throw a box around the dyn trait?

Reply:
yeah that's right! we can use a box to get around that compiler error (the memory size) i will give it a try and see how much code we can reduce!